plugins/resource_monitor_plugin/test/test_monitor_loop.cpp

```diff
@@ -0,0 +1,430 @@
+         }
+
+         i++;
+
+         return rc;
+      };
+
+      mock_get_stat = []( const char *path, struct stat *buf ) -> int {
+         buf->st_dev = 0;
+         return 0;
+      };
+
+      set_threshold(80);
+      set_sleep_time(interval);
+      add_file_system("/test");
+
+      auto start = std::chrono::system_clock::now();
+
+      auto monitor_thread = std::thread( [this] {
+         space_monitor_loop();
+         ctx.run();
+      });
+
+      monitor_thread.join();
+
+      auto end = std::chrono::system_clock::now();
+      std::chrono::duration<double> diff = end - start;
+
+      bool finished_in_time = (diff >= std::chrono::duration<double>(num_loops * interval) && diff <= std::chrono::duration<double>(num_loops * interval + 1)) ? true : false;
+
+      BOOST_TEST(finished_in_time == true);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(ten_loops_5_sec_interval, space_handler_fixture)
+   {
+      auto constexpr num_loops = 10;
+      auto constexpr interval  = 5;
+
+      mock_get_space = [ i = 0 ]( const bfs::path& p, boost::system::error_code& ec) mutable -> bfs::space_info {
+         ec = boost::system::errc::make_error_code(errc::success);
+
+         bfs::space_info rc;
+         rc.capacity  = 1000000;
+
+         if ( i < num_loops + 1 ) {  // "+ 1" for the get_space in add_file_system
+            rc.available = 300000;
+         } else {
+            rc.available = 100000;
+         }
+
+         i++;
+
+         return rc;
+      };
+
+      mock_get_stat = []( const char *path, struct stat *buf ) -> int {
+         buf->st_dev = 0;
+         return 0;
+      };
+
+      set_threshold(80);
+      set_sleep_time(interval);
+      add_file_system("/test");
+
+      auto start = std::chrono::system_clock::now();
+
+      auto monitor_thread = std::thread( [this] {
+         space_monitor_loop();
+         ctx.run();
+      });
+
+      monitor_thread.join();
+
+      auto end = std::chrono::system_clock::now();
+      std::chrono::duration<double> diff = end - start;
+
+      bool finished_in_time = (diff >= std::chrono::duration<double>(num_loops * interval) && diff <= std::chrono::duration<double>(num_loops * interval + 1)) ? true : false;
+
+      BOOST_TEST(finished_in_time == true);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(one_loop_59_sec_interval, space_handler_fixture)
+   {
+      auto constexpr num_loops = 1;
+      auto constexpr interval  = 59;
+
+      mock_get_space = [ i = 0 ]( const bfs::path& p, boost::system::error_code& ec) mutable -> bfs::space_info {
+         ec = boost::system::errc::make_error_code(errc::success);
+
+         bfs::space_info rc;
+         rc.capacity  = 1000000;
+
+         if ( i < num_loops + 1 ) {  // "+ 1" for the get_space in add_file_system
+            rc.available = 300000;
+         } else {
+            rc.available = 100000;
+         }
+
+         i++;
+
+         return rc;
+      };
+
+      mock_get_stat = []( const char *path, struct stat *buf ) -> int {
```

Comment:
This is repeated in each test. Please factor out all this copy/pasted code into something easier to maintain.